frame/evm/precompile/modexp/src/lib.rs

```diff
@@ -77,15 +78,19 @@ fn calculate_gas_cost(
 			//   must be > 0)
 			// * the addition can't overflow because the terms are both capped at roughly
 			//   8 * max size of exp_length (1024)
-			iteration_count =
-				(8 * (exp_length - 32)) + exponent.bitand(max_256_bit_uint).bits() - 1;
+			// * the EIP spec is written in python, in which (exponent & (2**256 - 1)) takes the
+			//   FIRST 32 bytes. However this `BigUint` `&` operator takes the LAST 32 bytes.
+			//   We thus instead take the bytes manually.
```

Comment:
This comment makes sense in the context of this PR (where the `&` operation goes away) but won't make sense without that context. In other words, this won't make sense without seeing that the `.bitand()` call is being removed.

Reply:
I think it makes sense to explain the confusing part of the python spec, to justify why we do differently.